roboteam_ai/src/skills/EnterFormation.cpp

```diff
@@ -0,0 +1,44 @@
+//
+// Created by mrlukasbos on 23-1-19.
+//
+
+#include "EnterFormation.h"
+#include "../utilities/Coach.h"
+#include "../control/ControlUtils.h"
+
+namespace rtt {
+namespace ai {
+
+EnterFormation::EnterFormation(std::string name, bt::Blackboard::Ptr blackboard) : Skill(std::move(name), std::move(blackboard)) {}
+
+void EnterFormation::onInitialize() {
+    coach::Coach::addFormationRobot(robot->id);
+}
+
+bt::Node::Status EnterFormation::onUpdate() {
+    auto robotPos = rtt::Vector2(robot->pos);
+    Vector2 targetLocation = coach::Coach::getFormationPosition(robot->id);
+    Vector2 targetToLookAtLocation = Field::get_their_goal_center();
+
+    roboteam_msgs::RobotCommand cmd;
+    cmd.id = robot->id;
+    cmd.use_angle = 1;
+
+    if (robotPos.dist(targetLocation) > 0.1) {
```

Comment:
I don't really like this design if I'm honest, because both getting into position and setting final angle. Also, goToPos already has a check like this in it (I believe with currently >0.3) so it will stop the velocities before it is actually there.